### PR TITLE
Whisper large-v3

### PR DESCRIPTION
**Pull request by flyingleafe**
Convert from HF format, including the feature extractor parameters.
Example: https://huggingface.co/flyingleafe/faster-whisper-large-v3

### CODE REVIEW

**Comment by jakariaemon** [comment]
Is that alignment head is correct in the cofig.json?

**Comment by souvikqb** [comment]
I saw a large-v3 implementation with faster_whisper (https://github.com/guillaumekln/faster-whisper/issues/547) but it's quite slow.

Large-v3 is very fast with batching as shown here --- https://huggingface.co/openai/whisper-large-v3

Batching speeds up the transcription process by a lot. The only reason I wish to use faster_whisper is cause it provides things like srt, verbose, word level transcription

**Comment by slavonnet** [comment]
i also foun need add
```
        if "large-v3" in model_size_or_path:
            self.feature_extractor.mel_filters = self.feature_extractor.get_mel_filters(self.feature_extractor.sampling_rate, self.feature_extractor.n_fft, n_mels=128)
```
to  model class 

and have 2 bugs
-  `The current model is English-only but the language parameter is set to 'ru'; using 'en' instead.
Detected language 'English' with probability 1.000000` but call is russian
-
```
Traceback (most recent call last):
  File "/home/venv/bin/whisper-ctranslate2", line 8, in <module>
    sys.exit(main())
  File "/home/venv/lib/python3.8/site-packages/src/whisper_ctranslate2/whisper_ctranslate2.py", line 560, in main
    result = transcribe.inference(
  File "/home/venv/lib/python3.8/site-packages/src/whisper_ctranslate2/transcribe.py", line 170, in inference
    for segment in segments:
  File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 945, in restore_speech_timestamps
    for segment in segments:
  File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 456, in generate_segments
    ) = self.generate_with_fallback(encoder_output, prompt, tokenizer, options)
  File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 664, in generate_with_fallback
    result = self.model.generate(
ValueError: <|startoftranscript|> token was not found in the prompt

````

**Comment by souvikqb** [comment]
> i also foun need add
> 
> ```
>         if "large-v3" in model_size_or_path:
>             self.feature_extractor.mel_filters = self.feature_extractor.get_mel_filters(self.feature_extractor.sampling_rate, self.feature_extractor.n_fft, n_mels=128)
> ```
> 
> to model class
> 
> and have 2 bugs
> 
> * `The current model is English-only but the language parameter is set to 'ru'; using 'en' instead. Detected language 'English' with probability 1.000000` but call is russian
> 
> ```
> Traceback (most recent call last):
>   File "/home/venv/bin/whisper-ctranslate2", line 8, in <module>
>     sys.exit(main())
>   File "/home/venv/lib/python3.8/site-packages/src/whisper_ctranslate2/whisper_ctranslate2.py", line 560, in main
>     result = transcribe.inference(
>   File "/home/venv/lib/python3.8/site-packages/src/whisper_ctranslate2/transcribe.py", line 170, in inference
>     for segment in segments:
>   File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 945, in restore_speech_timestamps
>     for segment in segments:
>   File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 456, in generate_segments
>     ) = self.generate_with_fallback(encoder_output, prompt, tokenizer, options)
>   File "/home/venv/lib/python3.8/site-packages/faster_whisper/transcribe.py", line 664, in generate_with_fallback
>     result = self.model.generate(
> ValueError: <|startoftranscript|> token was not found in the prompt
> ```

A little unrelated, but did you try large-v3 with batching or just normally?

**Comment by slavonnet** [comment]
after recompile from sources current build lang is set to russian but result in engels 

```

home/venv/bin/whisper-ctranslate2 /var/spool/asterisk//monitor/2023/11/09/q-1507-89370130386-20231109-153209-1699533129.18256-t.wav /var/spool/asterisk//monitor/2023/11/09/q-1507-89370130386-20231109-153209-1699533129.18256-r.wav --languag ru --vad_filter True --word_timestamps True --vad_threshold 0.9 --vad_min_speech_duration_ms 500 --vad_min_silence_duration_ms 3000 --initial_prompt 'Бигтелеком, Химки, Интернет, роутер,  Тариф, пин' --device cpu --best_of 1 --model_directory /home/venv/faster-whisper-large-v3 --condition_on_previous_text False --model_dir /home/venv/.mycache --local_files_only True --threads 4 --output_dir /tmp/tmp.fIkPsuX1sK -f all

File: '/var/spool/asterisk//monitor/2023/11/09/q-1507-89370130386-20231109-153209-1699533129.18256-t.wav'
Detected language 'Russian' with probability 1.000000
[00:00.690 --> 00:01.290]  Hello,
[00:02.070 --> 00:02.130]  Remember
[00:02.130 --> 00:02.510]  is
[00:02.510 --> 00:02.890]  do
[00:02.890 --> 00:09.660]  What is the question? What exactly?
[00:11.220 --> 00:12.020]  The fire, right?
[00:13.580 --> 00:15.740]  Well, yes, our equipment is disabled there, the break.
[00:16.380 --> 00:19.440]  Restore the line, as soon as it is restored, everything should turn on.
[00:19.840 --> 00:21.640]  They promised to do it by the end of the day.
[00:23.180 --> 00:23.820]  Please, all the best.
[00:25.320 --> 00:25.860]  Goodbye.

File: '/var/spool/asterisk//monitor/2023/11/09/q-1507-89370130386-20231109-153209-1699533129.18256-r.wav'
Detected language 'Russian' with probability 1.000000
[00:04.340 --> 00:04.940]  Hello,
[00:05.600 --> 00:08.320]  I don't
[00:10.360 --> 00:12.360]  Don't
[00:21.000 --> 00:22.500]  Don't
Transcription results written to '/tmp/tmp.fIkPsuX1sK' directory

real    6m29.154s


``

**Comment by bungerr** [comment]
see: progress in #548

File: faster_whisper/utils.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "large-v1": "guillaumekln/faster-whisper-large-v1",
     "large-v2": "guillaumekln/faster-whisper-large-v2",
     "large": "guillaumekln/faster-whisper-large-v2",
+    "large-v3": "flyingleafe/faster-whisper-large-v3",
```
Comment (ostegm) [inline faster_whisper/utils.py:24]: Did you convert this after the [pr fixing alignment heads was included in ctranslate2](https://github.com/OpenNMT/CTranslate2/pull/1530)? and if not, what is the expected impact?

**Comment by BBC-Esq** [comment]
Ctranslate2 has been successfully updated to support Whisper large-v3: 

https://github.com/OpenNMT/CTranslate2/releases/tag/v3.21.0

**Comment by flyingleafe** [comment]
Closing in favor of #548

**Pull request closed without merging**